### PR TITLE
Events tabs

### LINKED ISSUE #2902: [FEATURE] Events Mobile - filter UI improvement
### Description

Now that there are 4 options to filter events, they are beginning to take alot of space especially on mobile.  This ticket is to modify the filter for mobile to a dropdown, smaller buttons, or another idea (open to discussion)

Current UI
events page mobile

### Screenshots

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by Aadarsh805**
## Fixes Issue
Closes #2902 

## Changes proposed
Changed the events page layout to tabs format same as in user profile section and a select dropdown for mobile screen size

Created `EventTab.js` separate file for events tabs
implemented both tabs layout and select dropdown for large and smaller screen size

## Check List (Check all the applicable boxes) 

- [ x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [ x] All new and existing tests passed.
- [x ] This PR does not contain plagiarized content.
- [ x] The title of my pull request is a short description of the requested changes.

##### These were taking too much screen space
image

### After:
image

### video screen capture:
[screen-capture (9).webm](https://user-images.githubusercontent.com/95094057/211624580-5719c6aa-b34c-4d20-872a-5fa1ce7c63a3.webm)

Idk why it didn't capture the dropdown, but it is there you can see in this video down here 
[screen-capture (10).webm](https://user-images.githubusercontent.com/95094057/211624813-3ce41258-ec6d-4ad2-9d7e-61077f87e882.webm)

## Note to reviewers
I think the code can be cleaned up a little more, `UserTabs` and `EventTabs` pretty much do the same thing, maybe we can remove the reputation by using one for both

### CODE REVIEW

File: pages/events.js [hunk 1]
```diff
@@ -30,7 +29,54 @@ export default function Events({ events }) {
...
+      description: "You can submit a talk to this conference",
+      key: "cfpOpen",
+      icon: FaMicrophoneAlt,
+      current: false,
+    },
+    {
+      title: "In person",
+      description: "These are in person events",
+      key: "inPerson",
+      icon: MdOutlinePeople,
```
Comment (ChinmayMhatre) [inline pages/events.js:51]: I noticed these icons aren't used in the `EventTabs.js` the icons are hard coded in the file. Could you tell me where these are used?

File: pages/events.js [hunk 2]
```diff
@@ -30,7 +29,54 @@ export default function Events({ events }) {
...
+      key: "all",
+      icon: FaListUl,
+      current: true,
+    },
+    {
+      title: "CFP open",
+      description: "You can submit a talk to this conference",
+      key: "cfpOpen",
+      icon: FaMicrophoneAlt,
+      current: false,
```
Comment (ChinmayMhatre) [inline pages/events.js:45]: Instead of this could we have a state for the current tab? the value could be the `key` .

File: pages/events.js [hunk 3]
```diff
@@ -30,7 +29,54 @@ export default function Events({ events }) {
...
+      description: "Held virtually online event",
+      key: "virtual",
+      icon: MdOutlineOnlinePrediction,
+      current: false,
+    },
+  ];
+
+  let displayTabs = filters.flatMap((tab) => {
+    if (tab.key === "all") {
+      return { ...tab, total: categorisedEvents.all.length };
```
Comment (ChinmayMhatre) [inline pages/events.js:65]: This could be done directly in the `filters` object. the `if` statement seems unnecessary.

example 

```js
    {
      title: "Show all",
      description: "List all events with no filters",
      key: "all",
      icon: FaListUl,
      current: true,
      total: categorisedEvents.all.length
    }

 ```
and so on

**Review by ChinmayMhatre — CHANGES_REQUESTED** [review]
The UI looks good. Left some comments within the code which could be refactored

File: pages/events.js [hunk 4]
```diff
@@ -40,18 +86,36 @@ export default function Events({ events }) {
...
 
-        {!events.length && <Alert type="info" message="No events found" />}
-        <ul role="list" className="divide-y divide-gray-200">
-          {categorisedEvents[eventType].map((event, index) => (
-            <EventCard event={event} username={event.username} key={index} />
-          ))}
-        </ul>
+        {filters.map(
+          (filter) =>
+            tabs.find((tab) => tab.key === filter.key) &&
```
Comment (ChinmayMhatre) [inline pages/events.js:98]: This could be simplifies if we have a `currentTab` state

**Review by ChinmayMhatre — CHANGES_REQUESTED** [review]

**Review by ChinmayMhatre — CHANGES_REQUESTED** [review]
Please remove all the console.logs after debugging the code.

File: pages/events.js (hunk 4 above)
Comment (Aadarsh805) [reply pages/events.js:98]: This is similar to how it's being used in the `[username].js`, I'll try to simplify it more
thanks for all these suggestions

File: components/event/EventTabs.js [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+                    : "border-transparent text-gray-500 hover:text-gray-700 hover:border-gray-300",
+                  "w-1/4 py-4 px-1 text-center border-b-2 font-medium text-sm flex justify-center items-center gap-2 cursor-pointer"
+                )}
+                aria-current={tab.current ? "page" : undefined}
+              >
+                {tab.key === "all" && (
+                  <FaListUl title="Show All" className="text-base" />
+                )}
+                {tab.key === "virtual" && (
+                  <MdOutlineOnlinePrediction
```
[the remark this answers is not in the document]
Comment (Aadarsh805) [reply components/event/EventTabs.js:126]: Oh yeah, fixed this
image

File: pages/events.js (hunk 3 above)
Comment (Aadarsh805) [reply pages/events.js:65]: This works as good, a good one 👍
image

File: pages/events.js (hunk 1 above)
Comment (Aadarsh805) [reply pages/events.js:51]: These were being used in `EventKey.js` to create these cards 
image

**Comment by Aadarsh805** [comment]
@ChinmayMhatre hey, I'm having a problem with the sorting function, what happens is the sort works on the second click
it doesn't work on the first click but then it starts working after the first click
can you have a look in that functions code whenever you are free, thanks

File: pages/events.js (hunk 2 above)
Comment (Aadarsh805) [reply pages/events.js:45]: How would that work here?
image

Comment (ChinmayMhatre) [reply pages/events.js:46]: Since the `tabs` array of objects have `key` for each object. the state could be 

```jsx
const [eventType, setEventType] = useState("all");
```

initially. then the change function could fetch the `key` of the event type that the user click then set it to that.


the ui could be changed according to that

File: components/event/EventTabs.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (ChinmayMhatre) [reply components/event/EventTabs.js]: > @ChinmayMhatre hey, I'm having an issue with the sorting function, what happens is the sort works on the second click it doesn't work on the first click but then it starts working after the first click can you have a look at that functions code whenever you are free, thanks

I went Through the issue and noticed the issue was majorly discussing creating tabs instead of the four button.
I think we could focus on refactoring the tabs further and create a new issue for the sorting. You could take on that issue if you want since you had been working on this.

**Comment by ChinmayMhatre** [comment]
> @ChinmayMhatre hey, I'm having an issue with the sorting function, what happens is the sort works on the second click it doesn't work on the first click but then it starts working after the first click can you have a look at that functions code whenever you are free, thanks

I went Through the issue and noticed the issue was majorly discussing creating tabs instead of the four button.
I think we could focus on refactoring the tabs further and create a new issue for the sorting. You could take on that issue if you want since you had been working on this.

File: pages/events.js [hunk 6]
```diff
@@ -30,7 +29,47 @@ export default function Events({ events }) {
...
+      title: "Virtual",
+      description: "Held virtually online event",
+      key: "virtual",
+      icon: MdOutlineOnlinePrediction,
+      current: false,
+      total: categorisedEvents.virtual.length,
+    },
+  ];
+
+  let displayTabs = filters.flatMap((tab) => {
```
Comment (ChinmayMhatre) [inline pages/events.js:67]: Can't the `filters` variable be used directly? is `displayTabs` different? If not you could directly use `filter` or rename it appropriate if you like.

**Review by ChinmayMhatre — CHANGES_REQUESTED** [review]

File: pages/events.js (hunk 6 above)
Comment (Aadarsh805) [reply pages/events.js:67]: i tried using the filters directly but let me try again

**Comment by Aadarsh805** [comment]
resolved conflict in the above commit

File: pages/events.js (hunk 6 above)
Comment (Aadarsh805) [reply pages/events.js:67]: Oh it definitely works directly with `filters`, now I'm confused as why even was i doing the `flatMap` lol

**Comment by Aadarsh805** [comment]
> > > > @ChinmayMhatre hey, I'm having an issue with the sorting function, what happens is the sort works on the second click it doesn't work on the first click but then it starts working after the first click can you have a look at that functions code whenever you are free, thanks
> > > 
> > > 
> > > I went Through the issue and noticed the issue was majorly discussing creating tabs instead of the four button. I think we could focus on refactoring the tabs further in this issue and create a new issue for the sorting. You could take on that issue as well if you want since you had been working on this.
> > 
> > 
> > so I remove the sorting functionality and buttons for now ?
> 
> Yes we could add those later. It will be better if we add small changes in different Prs

so should i remove the sorting code completely or can i comment that out, as we might need it later

**Comment by ChinmayMhatre** [comment]
> > > > > @ChinmayMhatre hey, I'm having an issue with the sorting function, what happens is the sort works on the second click it doesn't work on the first click but then it starts working after the first click can you have a look at that functions code whenever you are free, thanks
> > > > 
> > > > 
> > > > I went Through the issue and noticed the issue was majorly discussing creating tabs instead of the four button. I think we could focus on refactoring the tabs further in this issue and create a new issue for the sorting. You could take on that issue as well if you want since you had been working on this.
> > > 
> > > 
> > > so I remove the sorting functionality and buttons for now ?
> > 
> > Yes we could add those later. It will be better if we add small changes in different Prs
> 
> so should i remove the sorting code completely or can i comment that out, as we might need it later

Remove the code from here. But maybe you can save it somewhere locally on your system for a reference

**Comment by Aadarsh805** [comment]
> > > > > > @ChinmayMhatre hey, I'm having an issue with the sorting function, what happens is the sort works on the second click it doesn't work on the first click but then it starts working after the first click can you have a look at that functions code whenever you are free, thanks
> > > > > 
> > > > > 
> > > > > I went Through the issue and noticed the issue was majorly discussing creating tabs instead of the four button. I think we could focus on refactoring the tabs further in this issue and create a new issue for the sorting. You could take on that issue as well if you want since you had been working on this.
> > > > 
> > > > 
> > > > so I remove the sorting functionality and buttons for now ?
> > > 
> > > 
> > > Yes we could add those later. It will be better if we add small changes in different Prs
> > 
> > 
> > so should i remove the sorting code completely or can i comment that out, as we might need it later
> 
> Remove the code from here. But maybe you can save it somewhere locally on your system for a reference

alright 🫡

**Comment by ChinmayMhatre** [comment]
@Aadarsh805 I have added a comment on how you could go about adding a state for the event type .Could you take a look?

**Comment by Aadarsh805** [comment]
> @Aadarsh805 I have added a comment on how you could go about adding a state for the event type .Could you take a look?

ohh yes yes we can do that

**Comment by ChinmayMhatre** [comment]
> > @Aadarsh805 I have added a comment on how you could go about adding a state for the event type .Could you take a look?
> 
> ohh yes yes we can do that

Try it out, let me know if you need any help

File: pages/events.js (hunk 6 above)
Comment (ChinmayMhatre) [reply pages/events.js]: Haha.. no worries

**Comment by Aadarsh805** [comment]
@ChinmayMhatre made a commit, have a look

**Comment by Aadarsh805** [comment]
> > @ChinmayMhatre made a commit, have a look
> 
> I'll take a look by EOD

sure sure whenever you are free

File: components/event/EventKey.js [hunk 7]
```diff
@@ -30,6 +30,7 @@ export default function EventKey({ categorisedEvents, onToggleEventType }) {
     },
   ];
 
+
```
Comment (ChinmayMhatre) [inline components/event/EventKey.js:33]: Space can be removed

File: components/user/UserTabs.js [hunk 8]
```diff
@@ -38,6 +38,7 @@ export default function UserTabs({ tabs, setTabs, setUserData, userData }) {
   const sortUserTabItems = (tabName, order) => {
     const { dataKey, sortKey } = getDataKeyAndSortKey(tabName);
     userData[dataKey].sort(function (a, b) {
+      console.log(userData, "boo");
```
Comment (ChinmayMhatre) [inline components/user/UserTabs.js:41]: Remove this console.log

File: pages/events.js [hunk 9]
```diff
@@ -40,19 +72,22 @@ export default function Events({ events }) {
       </Head>
       <Page>
         <h1 className="text-4xl mb-4 font-bold">Community events</h1>
-
-        <EventKey
-          categorisedEvents={categorisedEvents}
-          onToggleEventType={(newValue) => seteventType(newValue)}
+        <EventTabs
+          tabs={tabs}
+          setTabs={setTabs}
```
Comment (ChinmayMhatre) [inline pages/events.js:77]: I think `setTabs` isn't being used in the component. This can be removed

File: components/event/EventTabs.js [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+                    ? "border-indigo-500 text-indigo-600"
+                    : "border-transparent text-gray-500 hover:text-gray-700 hover:border-gray-300",
+                  "w-1/4 py-4 px-1 text-center border-b-2 font-medium text-sm flex justify-center items-center gap-2 cursor-pointer"
+                )}
+                aria-current={tab.key === eventType ? "page" : undefined}
+              >
+                {
+                  <tab.icon
+                    className={`${
+                      tab.key === "all" || tab.key === "cfpOpen"
```
Comment (ChinmayMhatre) [inline components/event/EventTabs.js:47]: Why the different sizes for the `all` and `cfpopen` types?

File: components/user/UserTabs.js [hunk 11]
```diff
@@ -87,17 +88,33 @@ export default function UserTabs({ tabs, setTabs, setUserData, userData }) {
...
+            defaultValue={tabs.find((tab) => tab.current).name}
+          >
+            {tabs.map((tab) => (
+              <option key={tab.name}>{tab.name}</option>
+            ))}
+          </select>
+        ) : (
+          <p className="block w-full rounded-md border-gray-300 focus:border-indigo-500 focus:ring-indigo-500 text-center">
+            <a
+              key={tabs[0].name}
```
Comment (ChinmayMhatre) [inline components/user/UserTabs.js:106]: I think the key can be `tabs[0].key` ?

File: components/user/UserTabs.js [hunk 12]
```diff
@@ -87,17 +88,33 @@ export default function UserTabs({ tabs, setTabs, setUserData, userData }) {
...
+        {tabs.length > 1 ? (
+          <select
+            id="tabs"
+            name="tabs"
+            onChange={changeTab}
+            className="block w-full rounded-md border-gray-300 focus:border-indigo-500 focus:ring-indigo-500"
+            defaultValue={tabs.find((tab) => tab.current).name}
+          >
+            {tabs.map((tab) => (
+              <option key={tab.name}>{tab.name}</option>
```
Comment (ChinmayMhatre) [inline components/user/UserTabs.js:100]: could the key be `tab.key` ?

**Review by ChinmayMhatre — CHANGES_REQUESTED** [review]
Hey @Aadarsh805  the changes look good, just have some minor changes remaining

File: components/event/EventTabs.js (hunk 10 above)
Comment (Aadarsh805) [reply components/event/EventTabs.js:47]: They are a little bigger compared to the other two, which looks odd

This is all icons having `text-xl`
image

and This is two of the first icons having `text-lg` and the other two `text-xl`
image

File: components/user/UserTabs.js (hunk 8 above)
Comment (Aadarsh805) [reply components/user/UserTabs.js:41]: done

**Comment by ChinmayMhatre** [comment]
@Aadarsh805 why was `userTabs` and `EventKey` deleted? I guess it is used somewhere, the github action failed. please check

**Comment by Aadarsh805** [comment]
> @Aadarsh805 why was `userTabs` and `EventKey` deleted? I guess it is used somewhere, the github action failed. please check

wait does it means it'll delete `userTabs` from the repo??

**Comment by ChinmayMhatre** [comment]
> > @Aadarsh805 why was `userTabs` and `EventKey` deleted? I guess it is used somewhere, the github action failed. please check
> 
> wait does it means it'll delete `userTabs` from the repo??

Yeah

File: components/event/EventTabs.js (hunk 10 above)
Comment (ChinmayMhatre) [reply components/event/EventTabs.js:47]: Alright seems fair

**Comment by ChinmayMhatre** [comment]
@Aadarsh805 if you want to revert the changes from the file you can you `git checkout main -- {name of the file} `
This commands takes that file from the main branch and copies it in this branch. Eddie made a video on this yesterday https://www.youtube.com/watch?v=YUHS5Ana7Jw

**Comment by Aadarsh805** [comment]
@ChinmayMhatre 
image

then this on pull
image

**Comment by Aadarsh805** [comment]
The delete was done through GitHub directly, that's why it wants me to pull first

**Comment by ChinmayMhatre** [comment]
@Aadarsh805 Maybe just pull the changes from the fork then
 revert back to this 0114d4185df5594c2ae945c72234a7371d0ab93 commit which is the one before things got deleted

**Comment by ChinmayMhatre** [comment]
@Aadarsh805 try not to make changes on github. Eddie has this video on how to keep the fork in sync and the workflow. You could check it out later. https://www.youtube.com/watch?v=vn2SgK4IbiM

**Comment by Aadarsh805** [comment]
> @Aadarsh805 try not to make changes on github. Eddie has this video on how to keep the fork in sync and the workflow. You could check it out later. https://www.youtube.com/watch?v=vn2SgK4IbiM

I'll check it out

now what 
image

**Comment by Aadarsh805** [comment]
It still saying to pull

**Comment by ChinmayMhatre** [comment]
> > @Aadarsh805 try not to make changes on github. Eddie has this video on how to keep the fork in sync and the workflow. You could check it out later. https://www.youtube.com/watch?v=vn2SgK4IbiM
> 
> I'll check it out
> 
> now what image

Are the files back locally?

**Comment by ChinmayMhatre** [comment]
> It still saying to pull

Did you pull the changes from your fork?

**Comment by Aadarsh805** [comment]
> > It still saying to pull
> 
> Did you pull the changes from your fork?

It's not letting me, should i pull with `--rebase`

**Comment by Aadarsh805** [comment]
Now I gotta do this for `EventKey.js` too, or should I leave it because it gets merged it won't require `EventKey.js`

**Comment by ChinmayMhatre** [comment]
> Why build fail

`Eventkeys` is also used in a file somewhere

**Comment by ChinmayMhatre** [comment]
> Now I gotta do this for `EventKey.js` too, or should I leave it because it gets merged it won't require `EventKey.js`

Revert it

**Comment by Aadarsh805** [comment]
> > Why build fail
> 
> `Eventkeys` is also used in a file somewhere

Oh yeah it is and it'll be even after this merge, totally my bad

**Comment by Aadarsh805** [comment]
Build successful 🥳
Omg this was scary XD
@ChinmayMhatre Thanks for all the help, really appreciate it 😇

**Comment by ChinmayMhatre** [comment]
@Aadarsh805 Why did you revert the `userTabs`? was it not related?

**Comment by Aadarsh805** [comment]
> userTabs

Yeah it was not related to this pr, I accidentally made changes to it copying the same functionality on `EventsTabs` ig

**Comment by Aadarsh805** [comment]
> > userTabs
> 
> Yeah it was not related to this pr, I accidentally made changes to it copying the same functionality on `EventsTabs` ig

Oh yes actually it was addressing another issue, which I didn't wanna add to this pr

**Review by ChinmayMhatre — APPROVED** [review]
The code looks good, functionality is working. Great work @Aadarsh805 . Let's gooo🚀🔥

**Comment by Aadarsh805** [comment]
> The code looks good, functionality is working. Great work @Aadarsh805 . Let's gooorocketfire

All thanks to you, thanks for all the suggestions and the help with Git commands

**Comment by ChinmayMhatre** [comment]
Hey @eddiejaoude just wanted to ping you, if you could take a look at this whenever you get time.

**Comment by eddiejaoude** [comment]
Thanks everyone, looks like great collaboration 👍 let me catch up on what I missed now

File: components/event/EventTabs.js [hunk 13]
```diff
@@ -0,0 +1,63 @@
+export function EventTabs({ tabs, eventType, setEventType }) {
+  const classNames = (...classes) => classes.filter(Boolean).join(" ");
+  const changeTab = (e, value) => {
+    setEventType(value);
```
Comment (eddiejaoude) [inline components/event/EventTabs.js:4]: I think you might need to `e.preventDefault()` here, we had a similar issue on the `UserTabs`

Have a look and see if you can keep it similar to those tabs `components/user/UserTabs.js`

File: pages/events.js [hunk 14]
```diff
@@ -40,19 +72,21 @@ export default function Events({ events }) {
       </Head>
       <Page>
         <h1 className="text-4xl mb-4 font-bold">Community events</h1>
-
-        <EventKey
-          categorisedEvents={categorisedEvents}
-          onToggleEventType={(newValue) => seteventType(newValue)}
+        <EventTabs
```
Comment (eddiejaoude) [inline pages/events.js:75]: If `EventKey` is not used anymore, we should remove the component

Comment (eddiejaoude) [reply pages/events.js:75]: Oh no wait it is used on the user's events tag

**Review by eddiejaoude — APPROVED** [review]
Looks good 👍 Great collaboration everyone 🔥

**Comment by eddiejaoude** [comment]
Oh I noticed when navigating between the tabs, the authors of the event change...

Screenshot 2023-01-18 at 11 05 54


After navigating...

Screenshot 2023-01-18 at 11 06 05

This is probably due to a `key` issue

**Review by eddiejaoude — CHANGES_REQUESTED** [review]
Bug found during testing

**Comment by Aadarsh805** [comment]
@eddiejaoude  what key would be the most appropriate here?

image


or we could add a new key variable using another map like this: 
```
{categorisedEvents[eventType]
            ?.map((event, i) => ({ ...event, id: i }))
            .map((event, index) => (
              <EventCard event={event} username={event.username} key={index} />
))}
```

**Comment by eddiejaoude** [comment]
We could always concatenate the event name and author - that will always be unique

**Review by eddiejaoude — APPROVED** [review]
Looks good 👍
